syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -40,6 +40,15 @@
 #include <getopt.h>
 #endif
 
+typedef struct _SubCommandDescriptor
+{
+  const gchar *mode;
+  const GOptionEntry *options;
+  const gchar *description;
+  gint (*main)(gint argc, gchar *argv[], const gchar *mode);
+  struct _SubCommandDescriptor *subcommands;
+} SubCommandDescriptor;
```

Comment:
I think this is a `CommandDescriptor`. Subcommand is just a `role` which is expressed by the variable name (`subcommands`).

Reply:
Agreed. Fixed